validator/keymanager/remote-web3signer/internal/client.go

```diff
@@ -0,0 +1,179 @@
+package internal
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common/hexutil"
+	"github.com/pkg/errors"
+	fieldparams "github.com/prysmaticlabs/prysm/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/crypto/bls"
+	"github.com/prysmaticlabs/prysm/encoding/bytesutil"
+)
+
+const (
+	ethApiNamespace = "/api/v1/eth2/sign/"
+	maxTimeout      = 3 * time.Second
+)
+
+type SignRequestJson []byte
+
+// HttpSignerClient defines the interface for interacting with a remote web3signer.
+type HttpSignerClient interface {
+	Sign(ctx context.Context, pubKey string, request SignRequestJson) (bls.Signature, error)
+	GetPublicKeys(ctx context.Context, url string) ([][48]byte, error)
+}
+
+// ApiClient a wrapper object around web3signer APIs. API docs found here https://consensys.github.io/web3signer/web3signer-eth2.html.
```

Comment:
Concerned about link becoming broken in the future. Maybe we can just say refer to the docs from Consensys' web3signer project